File: 229-majority-element-ii/majority-element-ii.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        count1 = 0
        count2 = 0
        element1 = -sys.maxsize - 1
        element2 = -sys.maxsize - 1

        for i in range(0,len(nums)):
            if(nums[i] == element1):
                count1+=1
            elif(nums[i] == element2):
                count2+=1
            elif(count1 == 0):
                element1 = nums[i]
                count1 = 1
            elif(count2 == 0):
                element2 = nums[i]
                count2 = 1
            else:
                count1-=1
                count2-=1

        ans = []
        maxNum = len(nums)//3
        count1 = 0
        count2 = 0
        for i in range (0,len(nums)):
            if(nums[i] == element1):
                count1+=1
            elif(nums[i] == element2):
                count2+=1

        if(count1 > maxNum): 
            ans.append(element1)
        if(count2 > maxNum): 
            ans.append(element2)

        return ans

File: 229-majority-element-ii/majority-element-ii.cpp
class Solution {
public:
    vector<int> majorityElement(vector<int>& nums) {
        int count1 = 0;
        int count2 = 0;
        int element1 = INT_MIN;
        int element2 = INT_MIN;

        for(int i=0;i<nums.size();i++)
        {
           if(nums[i] == element1)
           {
                count1++;
           }
           else if(nums[i] == element2)
           {
            count2++;
           }
           else if(count1 == 0)
           {
            element1 = nums[i];
            count1 = 1;
           }
           else if(count2 == 0)
           {
            element2 = nums[i];
            count2 = 1;
           }
           else{
            count1--;
            count2--;
           }
        }

        vector<int> ans;
        int maxNum = nums.size()/3;
        count1 = 0;
        count2 = 0;
        for(int i=0;i<nums.size();i++)
        {
            if(nums[i] == element1)
            {
                count1++;
            }
            else if(nums[i] == element2)
            {
                count2++;
            }
        }

        if(count1 > maxNum) ans.push_back(element1);
        if(count2 > maxNum) ans.push_back(element2);

        return ans;
    }
};